c7n/actions.py

```diff
@@ -186,26 +186,45 @@ class ModifyVpcSecurityGroupsAction(Action):
         'properties': {
             'type': {'enum': ['modify-security-groups']},
             'add': {'oneOf': [
-                {'type': 'string', 'pattern': '^sg-*'},
+                {'type': 'string'},
                 {'type': 'array', 'items': {
-                    'pattern': '^sg-*',
                     'type': 'string'}}]},
             'remove': {'oneOf': [
                 {'type': 'array', 'items': {
-                    'type': 'string', 'pattern': '^sg-*'}},
+                    'type': 'string'}},
                 {'enum': [
                     'matched', 'network-location', 'all',
-                    {'type': 'string', 'pattern': '^sg-*'}]}]},
+                    {'type': 'string'}]}]},
             'isolation-group': {'oneOf': [
-                {'type': 'string', 'pattern': '^sg-*'},
+                {'type': 'string'},
                 {'type': 'array', 'items': {
-                    'type': 'string', 'pattern': '^sg-*'}}]}},
+                    'type': 'string'}}]}},
         'anyOf': [
             {'required': ['isolation-group', 'remove', 'type']},
             {'required': ['add', 'remove', 'type']},
             {'required': ['add', 'type']}]
     }
 
+    def resolve_security_group_names(self, group_names):
+        """
+        Resolves Security Group names to Ids.
+        Returns an empty list if no matches
+        """
+        client = utils.local_session(
+            self.manager.session_factory).client('ec2')
+        filtered_sgs = client.describe_security_groups(
+            Filters=[
+                {
+                    'Name': 'group-name',
+                    'Values': group_names
+                }
+            ]
+        )['SecurityGroups']
+        filtered_ids = [a['GroupId'] for a in filtered_sgs]
+        if not filtered_ids:
+            self.log.warning("No Security Groups found for: %s" % group_names)
```

Comment:
what happens if one name is missing from a set?

Reply:
ah yea... describe_security_groups will only return the ones it can find. In this case I guess we should bail out if we can't find all the specified sg's